### ISSUE TITLE
OS X: No `mountpoint` for USB devices

### ISSUE THREAD

**Issue by mikew** [resin-io-modules/drivelist#75]
The OS X script ends up running `diskutil info /dev/disk2`, when it should be running `diskutil info /dev/disk2s1`. Using something like this is more reliable, but a bit slower

```bash
list_disks () {
	mounts=$(mount)
	diskutil list | while read line; do
		id=$(echo "$line" | rev | cut -d ' ' -f 1 | rev)
		if [[ "$id" == disk* ]]; then
			if [[ "$mounts" == *"/dev/$id on"* ]]; then
				echo "/dev/$id"
			fi
		fi
	done
}

for disk in $(list_disks); do
  # ...
```

**Comment by mikew** [comment]
Another way to write it to avoid both `mount` and `diskutil list` is:

```bash
list_disks () {
	mount | while read line; do
		dev=$(echo "$line" | cut -d ' ' -f 1)
		if [[ "$dev" == /dev/disk* ]]; then
			echo "$dev"
		fi
	done
}
```

**Comment by jviotti** [comment]
Hi @mikew ,

Your code means that we'll be outputting information about each partition as a separate object in the final array. The purpose of `drivelist` is to detect the connected drives (without caring about the partitions), so that's the reason we only call `diskutil info` for top level disks.

I'd be happy to take a PR that recursively scans the partitions of a disk, and creates a comma separated list of mount points for the top level disk.

**Comment by mikew** [comment]
The first function checks agains the device in the output of `mount`.

Before:

```
$ ./node_modules/drivelist/scripts/darwin.sh
device: /dev/disk0
description: APPLE SSD SM0256G
size: 251000193024
mountpoint:
name: /dev/disk0
system: True

device: /dev/disk1
description: APPLE SSD SM0256G
size: 199000412160
mountpoint: /
name: /dev/disk1
system: True

device: /dev/disk2
description: Ultra Fit
size: 62176362496
mountpoint:
name: /dev/disk2
system: False
```

After:

```
$ ./node_modules/drivelist/scripts/darwin.sh
device: /dev/disk1
description: APPLE SSD SM0256G
size: 199000412160
mountpoint: /
name: /dev/disk1
system: True

device: /dev/disk0s4
description: BOOTCAMP
size: 50778341376
mountpoint: /Volumes/BOOTCAMP
name: /dev/disk0s4
system: True

device: /dev/disk2s1
description: Untitled 1
size: 34359738368
mountpoint: /Volumes/ESD-USB
name: /dev/disk2s1
system: False
```

Note the proper mount point for the Bootcamp partition, and the mountpoint for the root and the USB. I get what you're saying, it's technically different, but this module on Windows got a mountpoint for the USB, and on OSX it didn't detect anything.

**Comment by jviotti** [comment]
Hey there @mikew ,

I see your point. The USB mount-point not being detected in OS X is definitely something we must address, but we do need to keep the main drives (e.g: `/dev/disk2` instead of `/dev/disk2s1`) being listed, at least in the main array.

A solution would be having a `partitions` array property in each drive object containing all this information.

For example:

```js
{
  device: '/dev/disk0',
  description: 'APPLE SSD SM0256G',
  size: 251000193024,
  name: '/dev/disk0',
  system: true,
  partitions: [
    {
      device: /dev/disk0s4
      description: BOOTCAMP
      size: 50778341376
      mountpoint: /Volumes/BOOTCAMP
    }
  ]
}
```

The problem is that this approach would lead to API incompatibility issues in Windows, since on that platform, each partitions gets assigned a different top level drive letter, therefore the `partitions` array might not make sense there and users would have to write code to handle these edge cases.

I believe the approach I suggested before should work for you, and it's reasonably easy to implement without having to deal with different OSes incompatibilities. So to summarise:

- Iterate through the top level disks (`/dev/disk0`, `/dev/disk1`, etc).
- If we can find a mount point for the top level disk, add it to `mountpoint` .
- Iterate through the disk's partitions, if any, and if a mount point is detected, append it to `mountpoint`.

The result would look something like this:

```yaml
device: /dev/disk0
description: APPLE SSD SM0256G
size: 251000193024
mountpoint: /Volumes/BOOTCAMP
name: /dev/disk0
system: True

device: /dev/disk2
description: Ultra Fit
size: 62176362496
mountpoint: /Volumes/ESD-USB
name: /dev/disk2
system: False
```

This of course discards some information like the partition name, and the partition size, but I believe it works for your use case, and we can take it from there to solve the partitions issue in a more elegant way.

What do you think?

**Comment by jviotti** [comment]
This issue was fixed in https://github.com/resin-io-modules/drivelist/pull/93, and released as v3.3.2 to NPM.
